server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -845,6 +863,21 @@ public boolean isPartialSearchableSnapshot() {
         return isPartialSearchableSnapshot;
     }
 
+    @Nullable
+    public IndexMode getIndexMode() {
```

Comment:
Wants javadoc I think.

Reply:
Just because this class is, like, *everywhere* and it's convenient to be able to see it on mouseover and the like.

Reply:
I will add javadoc.